File: python/zigzag_convert.py
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        num_iter = 2*numRows - 2
        output = ""
        row_cha = ["" for x in range(numRows)]
        if num_iter > 0:
            for i, i_cha in enumerate(s):
                if i%num_iter < numRows:
                    row_cha[i%num_iter] += i_cha
                else:
                    row_cha[num_iter - i%num_iter] += i_cha
                    
            for j, j_string in enumerate(row_cha):
                output += j_string
        else:
            output = s
            
        return output

File: c++/ZigZagConvert.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows <= 1)
            return s;
        
        int numIter = 2*numRows - 2;
        const int len = (int)s.length();
        string *str = new string[numRows];
        
        for (int i = 0; i < len; ++i){
            int rowIndex = i%numIter;
            if ( rowIndex < numRows)
                str[rowIndex].push_back(s[i]);
            else
                str[numIter - rowIndex].push_back(s[i]);
        }
        s.clear();
        for (int j = 0; j < numRows; ++j){
            s.append(str[j]);
        }
        
        delete[] str;
        return s;
    }
};